### PR TITLE
Chore: Update package versions when creating new Github Release

### LINKED ISSUE #427: Update package versions when creating new Github Release
**Description**

When creating a Github release, such as [stg-v3.1.1](https://github.com/dzcode-io/dzcode.io/releases/tag/stg-v3.1.1), we should update all `**/package.json` version fields to be the same version of that of the Github Release.

**Check List**

1. [x] add a script to `@dzcode.io/tooling` that takes a version as an argument and applies it across the repo's `pakcage.json` files
2. [x] modify the deploy workflows to run this script passing the `{{ github.ref_name }}` as argument.
3. [x] address #414 
4. [x] do the same as point (3) but for `./mobile`
5. [x] specify [Sentry release](https://docs.sentry.io/product/releases/) for all apps, take it from `package.json`.`version`

**Additional Comments**

This will be helpful, for tracking progress, error logs, and overall making the monorepo's version consistent

### PR DESCRIPTION
**Pull request by ZibanPirate**
# Description

With these changes, whenever we create a release, the deploy workflow will update all the package.json files as well as ./mobile/app.json` file with the newly created version (from Github Releases)

fixes #427 first and second tasks

- update repo versions on each production release
- added `lint:fix` script and fixed couple of lint errors while at it

## Type of change

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/tooling/package.json [hunk 1]
```diff
@@ -25,10 +26,25 @@
     "npm-run-all": "^4.1.5",
     "prettier": "^2.4.1",
     "rimraf": "^3.0.2",
+    "semver": "^7.3.5",
     "ts-jest": "^26.5.6",
     "ts-node": "^10.4.0",
     "typescript": "^4.5.2"
   },
   "license": "MIT",
-  "private": true
```
Comment (Souhaib-Benbouzid) [inline packages/tooling/package.json:33]: why did you remove the private attribute?

Comment (ZibanPirate) [reply packages/tooling/package.json:33]: actually i didn't, it's just got moved to line 41

**Review by Souhaib-Benbouzid — APPROVED** [review]
